File: src/main/java/org/example/kotlinWorkshop/java/lesson03/t2_map/t2_1/Main.java
package org.example.kotlinWorkshop.java.lesson03.t2_map.t2_1;

import java.util.Map;

public class Main {

  public static void main(String[] args) {
    var classWithImmutableMap = new ClassWithImmutableMap(
        Map.of("some_key", "some_value", "another_key", "another_value")
    );

    classWithImmutableMap.getMap().put("another_key3", "anotherValue");

    //unreachable code
    System.out.println(classWithImmutableMap.getMap());
    System.out.println(classWithImmutableMap.getMap().get("some_key"));
  }
}

File: src/main/kotlin/org/example/kotlinWorkshop/kotlin/lesson03/t2_map/t2_1/Main.kt
package org.example.kotlinWorkshop.kotlin.lesson03.t2_map.t2_1

fun main() {
    val classWithImmutableMap = ClassWithImmutableMap(
        mapOf("some_key" to "some_value", "another_key" to "another_value")
    )

    //do not exist
    //classWithImmutableMap.map.put("another_new_key", "new_value")
    //classWithImmutableMap.map.["another_new_key"] =  "new_value"

    println(classWithImmutableMap.map)
    println(classWithImmutableMap.map.get("some_key"))
    println(classWithImmutableMap.map["another_key"])
}